src/transformers/generation/utils.py

```diff
@@ -3975,8 +4007,13 @@ def _assisted_decoding(
             cur_len = input_ids.shape[-1]
 
             #  1. Fetch candidate sequences from a `CandidateGenerator`
-            candidate_input_ids, candidate_logits = candidate_generator.get_candidates(input_ids)
-            candidate_input_ids = candidate_input_ids.to(self.device)
+            if isinstance(candidate_generator, AssistedCandidateGeneratorDifferentTokenizers):
+                candidate_input_ids, candidate_logits = candidate_generator.get_candidates(
+                    input_ids, stopping_criteria
+                )
+            else:
+                candidate_input_ids, candidate_logits = candidate_generator.get_candidates(input_ids)
+
```

Comment:
I think we can avoid these changes: you are passing `stopping_criteria` to reach its `max_length` attribute.

If I'm not mistaken, the assistant model's `generation_config` should the same value for `max_length`

Reply:
good catch